### ISSUE TITLE
The zoom button is always visible

### ISSUE THREAD

**Issue by kyolser** [amcharts/amcharts4#2086]
Hey. The zoom button is visible when loading a graph. But it is not needed, because the scale is correct. How to fix this? How to show the button only when it is needed?
![cec5c219b6](https://user-images.githubusercontent.com/39780126/73537267-1f77f000-4439-11ea-96f5-a445d6d46865.jpg)
This happens if there is little data on the chart. If there is a lot of data, then everything works correctly
![7f7a78d2fe](https://user-images.githubusercontent.com/39780126/73537364-551cd900-4439-11ea-821e-01ae296aeaf3.jpg)

**Comment by zeroin** [comment]
Could you provide a sample codepen of this?

**Comment by kyolser** [comment]
am4core.ready(function() {
    am4core.useTheme(am4themes_spiritedaway);
    am4core.useTheme(am4themes_animated);

    var chart = am4core.create("graph", am4charts.XYChart);
    chart.paddingRight = 20;

    chart.data = [];
    for (key in dataWeightGraph) {
        chart.data.push({
            date: new Date(dataWeightGraph[key].date * 1000),
            value: dataWeightGraph[key].value
        });
    };

    chart.language.locale = am4lang_ru_RU;
    var dateAxis = chart.xAxes.push(new am4charts.DateAxis());
    dateAxis.renderer.grid.template.location = 0;
    dateAxis.minZoomCount = 5;

    var valueAxis = chart.yAxes.push(new am4charts.ValueAxis());
    valueAxis.title.text = "Вес, кг";
    valueAxis.title.rotation = -90;
    valueAxis.title.align = "left";
    valueAxis.title.valign = "top";
    valueAxis.title.dy = -10;
    valueAxis.title.fontWeight = 600;
    valueAxis.extraMin = 0.1;
    valueAxis.extraMax = 0.1;

    var series = chart.series.push(new am4charts.LineSeries());
    series.dataFields.dateX = "date";
    series.dataFields.valueY = "value";

    series.stroke = am4core.color("#84BB71"); 
    series.strokeWidth = 3;
    series.minBulletDistance = 15;

    series.tooltipText = "{valueY}";
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.background.fillOpacity = 0.5;
    series.tooltip.background.cornerRadius = 10;
    series.tooltip.background.strokeOpacity = 0;
    series.tooltip.background.fillOpacity = 1;
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.label.minWidth = 30;
    series.tooltip.label.minHeight = 30;
    series.tooltip.label.textAlign = "middle";
    series.tooltip.label.textValign = "middle";
    series.tooltipText = '{value}[/] кг'
    series.tooltip.getFillFromObject = false;
    series.tooltip.background.fill = am4core.color("#84BB71");

    var scrollbarX = new am4core.Scrollbar();
    chart.scrollbarX = scrollbarX;
    chart.scrollbarX.parent = chart.bottomAxesContainer;
    chart.scrollbarX.background.fill = am4core.color("#eee");
    chart.scrollbarX.endGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.endGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.icon.disabled = true;
    chart.scrollbarX.startGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.startGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.icon.disabled = true;
    chart.scrollbarX.thumb.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.thumb.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.thumb.background.states.getKey('down').properties.fill = am4core.color("#67A660");

    var bullet = series.bullets.push(new am4charts.CircleBullet());
    bullet.circle.strokeWidth = 2;
    bullet.circle.radius = 4;
    bullet.circle.fill = am4core.color("#fff");
    var bullethover = bullet.states.create("hover");
    bullethover.properties.scale = 1.3;

    chart.cursor = new am4charts.XYCursor();
    chart.cursor.behavior = "panXY";
    chart.cursor.xAxis = dateAxis;
    chart.cursor.snapToSeries = series;
    chart.cursor.lineX.stroke = am4core.color("#67A660");
    chart.cursor.lineX.strokeWidth = 2;
    chart.cursor.lineX.strokeOpacity = 0.5;
    chart.cursor.lineX.strokeDasharray = "";
    chart.cursor.lineY.stroke = am4core.color("#67A660");
    chart.cursor.lineY.strokeWidth = 2;
    chart.cursor.lineY.strokeOpacity = 0.5;
    chart.cursor.lineY.strokeDasharray = "";

    chart.zoomOutButton.background.cornerRadius(5, 5, 5, 5);
    chart.zoomOutButton.background.fill = am4core.color("#84BB71");
    chart.zoomOutButton.icon.stroke = am4core.color("#fff");
    chart.zoomOutButton.icon.strokeWidth = 2;
    chart.zoomOutButton.background.states.getKey("hover").properties.fill = am4core.color("#67A660");
    chart.zoomOutButton.background.states.getKey("down").properties.fill = am4core.color("#67A660");

});

**Comment by kyolser** [comment]
`am4core.ready(function() {
    am4core.useTheme(am4themes_spiritedaway);
    am4core.useTheme(am4themes_animated);

    var chart = am4core.create("graph", am4charts.XYChart);
    chart.paddingRight = 20;

    chart.data = [];
    for (key in dataWeightGraph) {
        chart.data.push({
            date: new Date(dataWeightGraph[key].date * 1000),
            value: dataWeightGraph[key].value
        });
    };

    chart.language.locale = am4lang_ru_RU;
    var dateAxis = chart.xAxes.push(new am4charts.DateAxis());
    dateAxis.renderer.grid.template.location = 0;
    dateAxis.minZoomCount = 5;

    var valueAxis = chart.yAxes.push(new am4charts.ValueAxis());
    valueAxis.title.text = "Вес, кг";
    valueAxis.title.rotation = -90;
    valueAxis.title.align = "left";
    valueAxis.title.valign = "top";
    valueAxis.title.dy = -10;
    valueAxis.title.fontWeight = 600;
    valueAxis.extraMin = 0.1;
    valueAxis.extraMax = 0.1;

    var series = chart.series.push(new am4charts.LineSeries());
    series.dataFields.dateX = "date";
    series.dataFields.valueY = "value";

    series.stroke = am4core.color("#84BB71"); 
    series.strokeWidth = 3;
    series.minBulletDistance = 15;

    series.tooltipText = "{valueY}";
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.background.fillOpacity = 0.5;
    series.tooltip.background.cornerRadius = 10;
    series.tooltip.background.strokeOpacity = 0;
    series.tooltip.background.fillOpacity = 1;
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.label.minWidth = 30;
    series.tooltip.label.minHeight = 30;
    series.tooltip.label.textAlign = "middle";
    series.tooltip.label.textValign = "middle";
    series.tooltipText = '{value}[/] кг'
    series.tooltip.getFillFromObject = false;
    series.tooltip.background.fill = am4core.color("#84BB71");

    var scrollbarX = new am4core.Scrollbar();
    chart.scrollbarX = scrollbarX;
    chart.scrollbarX.parent = chart.bottomAxesContainer;
    chart.scrollbarX.background.fill = am4core.color("#eee");
    chart.scrollbarX.endGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.endGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.icon.disabled = true;
    chart.scrollbarX.startGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.startGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.icon.disabled = true;
    chart.scrollbarX.thumb.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.thumb.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.thumb.background.states.getKey('down').properties.fill = am4core.color("#67A660");

    var bullet = series.bullets.push(new am4charts.CircleBullet());
    bullet.circle.strokeWidth = 2;
    bullet.circle.radius = 4;
    bullet.circle.fill = am4core.color("#fff");
    var bullethover = bullet.states.create("hover");
    bullethover.properties.scale = 1.3;

    chart.cursor = new am4charts.XYCursor();
    chart.cursor.behavior = "panXY";
    chart.cursor.xAxis = dateAxis;
    chart.cursor.snapToSeries = series;
    chart.cursor.lineX.stroke = am4core.color("#67A660");
    chart.cursor.lineX.strokeWidth = 2;
    chart.cursor.lineX.strokeOpacity = 0.5;
    chart.cursor.lineX.strokeDasharray = "";
    chart.cursor.lineY.stroke = am4core.color("#67A660");
    chart.cursor.lineY.strokeWidth = 2;
    chart.cursor.lineY.strokeOpacity = 0.5;
    chart.cursor.lineY.strokeDasharray = "";

    chart.zoomOutButton.background.cornerRadius(5, 5, 5, 5);
    chart.zoomOutButton.background.fill = am4core.color("#84BB71");
    chart.zoomOutButton.icon.stroke = am4core.color("#fff");
    chart.zoomOutButton.icon.strokeWidth = 2;
    chart.zoomOutButton.background.states.getKey("hover").properties.fill = am4core.color("#67A660");
    chart.zoomOutButton.background.states.getKey("down").properties.fill = am4core.color("#67A660");

});
`

**Comment by kyolser** [comment]
```
am4core.ready(function() {
    am4core.useTheme(am4themes_spiritedaway);
    am4core.useTheme(am4themes_animated);
    var chart = am4core.create("graph", am4charts.XYChart);
    chart.paddingRight = 20;

    chart.data = [];
    for (key in dataWeightGraph) {
        chart.data.push({
            date: new Date(dataWeightGraph[key].date * 1000),
            value: dataWeightGraph[key].value
        });
    };

    chart.language.locale = am4lang_ru_RU;
    var dateAxis = chart.xAxes.push(new am4charts.DateAxis());
    dateAxis.renderer.grid.template.location = 0;
    dateAxis.minZoomCount = 5;

    var valueAxis = chart.yAxes.push(new am4charts.ValueAxis());
    valueAxis.title.text = "Вес, кг";
    valueAxis.title.rotation = -90;
    valueAxis.title.align = "left";
    valueAxis.title.valign = "top";
    valueAxis.title.dy = -10;
    valueAxis.title.fontWeight = 600;
    valueAxis.extraMin = 0.1;
    valueAxis.extraMax = 0.1;

    var series = chart.series.push(new am4charts.LineSeries());
    series.dataFields.dateX = "date";
    series.dataFields.valueY = "value";

    series.stroke = am4core.color("#84BB71");
    series.strokeWidth = 3;
    series.minBulletDistance = 15;

    series.tooltipText = "{valueY}";
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.background.fillOpacity = 0.5;
    series.tooltip.background.cornerRadius = 10;
    series.tooltip.background.strokeOpacity = 0;
    series.tooltip.background.fillOpacity = 1;
    series.tooltip.pointerOrientation = "vertical";
    series.tooltip.label.minWidth = 30;
    series.tooltip.label.minHeight = 30;
    series.tooltip.label.textAlign = "middle";
    series.tooltip.label.textValign = "middle";
    series.tooltipText = '{value}[/] кг'
    series.tooltip.getFillFromObject = false;
    series.tooltip.background.fill = am4core.color("#84BB71");

    var scrollbarX = new am4core.Scrollbar();
    chart.scrollbarX = scrollbarX;
    chart.scrollbarX.parent = chart.bottomAxesContainer;
    chart.scrollbarX.background.fill = am4core.color("#eee");
    chart.scrollbarX.endGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.endGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.endGrip.icon.disabled = true;
    chart.scrollbarX.startGrip.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.startGrip.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.background.states.getKey('down').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.startGrip.icon.disabled = true;
    chart.scrollbarX.thumb.background.fill = am4core.color("#84BB71");
    chart.scrollbarX.thumb.background.states.getKey('hover').properties.fill = am4core.color("#67A660");
    chart.scrollbarX.thumb.background.states.getKey('down').properties.fill = am4core.color("#67A660");

    var bullet = series.bullets.push(new am4charts.CircleBullet());
    bullet.circle.strokeWidth = 2;
    bullet.circle.radius = 4;
    bullet.circle.fill = am4core.color("#fff");
    var bullethover = bullet.states.create("hover");
    bullethover.properties.scale = 1.3;

    chart.cursor = new am4charts.XYCursor();
    chart.cursor.behavior = "panXY";
    chart.cursor.xAxis = dateAxis;
    chart.cursor.snapToSeries = series;
    chart.cursor.lineX.stroke = am4core.color("#67A660");
    chart.cursor.lineX.strokeWidth = 2;
    chart.cursor.lineX.strokeOpacity = 0.5;
    chart.cursor.lineX.strokeDasharray = "";
    chart.cursor.lineY.stroke = am4core.color("#67A660");
    chart.cursor.lineY.strokeWidth = 2;
    chart.cursor.lineY.strokeOpacity = 0.5;
    chart.cursor.lineY.strokeDasharray = "";

    chart.zoomOutButton.background.cornerRadius(5, 5, 5, 5);
    chart.zoomOutButton.background.fill = am4core.color("#84BB71");
    chart.zoomOutButton.icon.stroke = am4core.color("#fff");
    chart.zoomOutButton.icon.strokeWidth = 2;
    chart.zoomOutButton.background.states.getKey("hover").properties.fill = am4core.color("#67A660");
    chart.zoomOutButton.background.states.getKey("down").properties.fill = am4core.color("#67A660");

});
```

**Comment by kyolser** [comment]
https://codepen.io/kyolser/pen/KKwjRww

**Comment by zeroin** [comment]
This problem was fixed some versions ago, if you change 4.8.0 to the most recent, the problem should be gone.

**Comment by kyolser** [comment]
Thanks, it worked.
